File: Code/Components/DanpheEMR.ServerModel/AccountingModels/ChartOfAccountModel.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DanpheEMR.ServerModel
{
    public class ChartOfAccountModel
    {
        [Key]
        public int ChartOfAccountId { get; set; }
        public string ChartOfAccountName { get; set; }
        public int? PrimaryGroupId { get; set; }
        public string COACode { get; set; }
        public string Description { get; set; }
        public DateTime? CreatedOn { get; set; }
        public DateTime? ModifiedOn { get; set; }
        public int? ModifiedBy { get; set; }
        public int? CreatedBy { get; set; }
        public bool? IsActive { get; set; }
    }
}

File: Code/Websites/DanpheEMR/wwwroot/DanpheApp/src/app/accounting/settings/shared/chart-of-account.model.ts
import { FormBuilder, FormGroup, Validators } from "@angular/forms";
import * as moment from "moment";
import { ENUM_DateTimeFormat } from "../../../shared/shared-enums";

export class ChartofAccountModel {
    public ChartOfAccountId: number = 0;
    public ChartOfAccountName: string = "";
    public PrimaryGroupId: number = 0;
    public COACode: string = null;
    public Description: string = null;
    public CreatedBy: number = 1;
    public CreatedOn: string = null;
    public ModifiedBy: number = 0;
    public ModifiedOn: string = null;
    public IsActive: boolean = true;
    public COAValidator: FormGroup = null;
    constructor() {
        this.CreatedOn = moment().format(ENUM_DateTimeFormat.Year_Month_Day_Hour_Minute);
        var _formBuilder = new FormBuilder();
        this.COAValidator = _formBuilder.group({
            'COAName': ['', Validators.compose([Validators.required])],
            'primaryGroup': ['', Validators.compose([Validators.required])],
        });
    }

    public IsDirty(fieldName): boolean {
        if (fieldName == undefined)
            return this.COAValidator.dirty;
        else
            return this.COAValidator.controls[fieldName].dirty;
    }

    public IsValid(): boolean { if (this.COAValidator.valid) { return true; } else { return false; } } public IsValidCheck(fieldName, validator): boolean {
        if (fieldName == undefined) {
            return this.COAValidator.valid;

        }

        else
            return !(this.COAValidator.hasError(validator, fieldName));
    }
}